lib/rules/prefer-destructuring.js

```diff
@@ -110,22 +136,21 @@ module.exports = {
             }
 
             if (isArrayIndexAccess(rightNode)) {
-                if (checkArrays) {
+                if ((checkArraysInVariableDeclarator && reportNode.type === "VariableDeclarator") || (checkArraysInAssignmentExpression && reportNode.type === "AssignmentExpression")) {
                     report(reportNode, "array");
                 }
                 return;
             }
 
-            if (checkObjects && enforceForRenamedProperties) {
+            if ((checkObjectsInVariableDeclarator || checkObjectsInAssignmentExpression) && enforceForRenamedProperties) {
                 report(reportNode, "object");
                 return;
             }
 
-            if (checkObjects) {
+            if ((checkObjectsInVariableDeclarator && reportNode.type === "VariableDeclarator") || (checkObjectsInAssignmentExpression && reportNode.type === "AssignmentExpression")) {
```

Comment:
I think the value in this `if` statement is redundant, because `reportNode.type` will always be either `"VariableDeclarator"` or `"AssignmentExpression"`. It can be simplified to

```js
if (checkObjectsInVariableDeclarator || checkObjectsInAssignmentExpression) {
  // ...
}
```